ops/services/alerts/app_service_metrics/exceptions.tf

```diff
@@ -6,18 +6,10 @@ resource "azurerm_monitor_diagnostic_setting" "collect_appserviceconsolelogs" {
 
   enabled_log {
     category = "AppServiceConsoleLogs"
-    retention_policy {
```

Comment:
To clarify... are we removing this because it [is deprecated](https://github.com/hashicorp/terraform-provider-azurerm/pull/23260)? 🤔 

And are we not replacing it with `azurerm_storage_management_policy` as suggested because it is [being deprecated in v4](https://github.com/hashicorp/terraform-provider-azurerm/pull/27096/files#diff-87d772292cdc1025fcb86402d5076b0dd75bae080bbb78b55feff3c4467bf10cL127)? See also (https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/migrate-to-azure-storage-lifecycle-policy?tabs=portal)

With this being removed, is there any way to configure this or are we relying on some defaults somewhere? 👀

Reply:
This is a good callout. We can't change retention policies for logs without threatening our ATO. Definitely need to make sure we're accounting for it somewhere.

Reply:
Due to this PR only upgrading `terraform` and `setup-terraform` and not the azure provider, we're fine because we're still using `hashicorp/azurerm 3.100.0`. In the PR I'll push out after this one, I'll address all of the deprecated attributes. The attributes that come up as not supported would affect this PR

In the [ticket](https://app.zenhub.com/workspaces/devsecops-63d830363b26890016306a3f/issues/gh/cdcgov/prime-simplereport/7235) for the azurerm provider upgrade, I added your comment in the details so I'll make sure its included and verify it stays within the requirements of the ATO.  Thank you both for bringing this to my attention, good catch.

Reply:
I'm not so sure that logic makes sense. If you change the code here, you will change the retention policy, regardless of whether you are just intending to upgrade provider versions. 

Take a look at your most recent deploy: https://github.com/CDCgov/prime-simplereport/actions/runs/10953277197/job/30413331322#step:3:1

The `terraform apply` phase shows multiple places where we can see retention adjustments. There are also places where OMHS-placed tags are getting blown away. I don't think these changes can be propagated up to Prod in their current state.

Can we figure out what's going on here before we proceed further?

Reply:
I misunderstood, and agree with waiting to push this forward. The resource @emyl3 referenced has to be added and figure out why the OMHS-placed tags are being removed during terraform apply.